From bf5f04ef535c7bcdaeae955126ee1734e298cc10 Mon Sep 17 00:00:00 2001
From: Austin Mayes <austin@avicus.net>
Date: Tue, 16 Jul 2019 16:52:28 -0500
Subject: [PATCH] Implement common


diff --git a/pom.xml b/pom.xml
index ee562d2e..abfe2e4a 100644
--- a/pom.xml
+++ b/pom.xml
@@ -92,6 +92,11 @@
             <version>1.3</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>network.stratus</groupId>
+            <artifactId>common</artifactId>
+            <version>1.0-SNAPSHOT</version>
+        </dependency>
     </dependencies>
 
     <!-- required until fixed plexus-compiler-eclipse is deployed -->
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index e18da962..79c7fe10 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -15,7 +15,7 @@ import java.util.UUID;
 import net.md_5.bungee.api.chat.BaseComponent;
 import net.minecraft.server.*;
 
-import network.stratus.sportpaper.api.text.PersonalizedComponent;
+import network.stratus.common.text.PersonalizedComponent;
 import org.apache.commons.lang.Validate;
 import org.bukkit.BlockChangeDelegate;
 import org.bukkit.Bukkit;
diff --git a/src/main/java/org/bukkit/craftbukkit/command/ProxiedNativeCommandSender.java b/src/main/java/org/bukkit/craftbukkit/command/ProxiedNativeCommandSender.java
index be7f7d8e..ac02ba80 100644
--- a/src/main/java/org/bukkit/craftbukkit/command/ProxiedNativeCommandSender.java
+++ b/src/main/java/org/bukkit/craftbukkit/command/ProxiedNativeCommandSender.java
@@ -59,11 +59,6 @@ public class ProxiedNativeCommandSender implements ProxiedCommandSender {
         return getCallee().getName();
     }
 
-    @Override
-    public String getName(CommandSender viewer) {
-        return getCallee().getName(viewer);
-    }
-
     @Override
     public boolean isPermissionSet(String name) {
         return getCaller().isPermissionSet(name);
diff --git a/src/main/java/org/bukkit/craftbukkit/command/ServerCommandSender.java b/src/main/java/org/bukkit/craftbukkit/command/ServerCommandSender.java
index c32dee20..a0f3f1d5 100644
--- a/src/main/java/org/bukkit/craftbukkit/command/ServerCommandSender.java
+++ b/src/main/java/org/bukkit/craftbukkit/command/ServerCommandSender.java
@@ -140,7 +140,57 @@ public abstract class ServerCommandSender implements CommandSender {
     }
 
     @Override
-    public String getName(CommandSender viewer) {
-        return this.getName();
+    public String getName(network.stratus.common.CommandSender viewer) {
+        return getName();
+    }
+
+    @Override
+    public void setFakeName(network.stratus.common.CommandSender viewer, String name) {
+
+    }
+
+    @Override
+    public String getFakeName(network.stratus.common.CommandSender viewer) {
+        return getName();
+    }
+
+    @Override
+    public boolean hasFakeName(network.stratus.common.CommandSender viewer) {
+        return false;
+    }
+
+    @Override
+    public void clearFakeNames() {
+
+    }
+
+    @Override
+    public void setFakeDisplayName(network.stratus.common.CommandSender viewer, String name) {
+
+    }
+
+    @Override
+    public String getFakeDisplayName(network.stratus.common.CommandSender viewer) {
+        return getName();
+    }
+
+    @Override
+    public boolean hasFakeDisplayName(network.stratus.common.CommandSender viewer) {
+        return false;
+    }
+
+    @Override
+    public void clearFakeDisplayNames() {
+
+    }
+
+    @Override
+    public String getPlayerListName(network.stratus.common.CommandSender viewer) {
+        return getName();
+    }
+
+    @Override
+    public String getDisplayName(network.stratus.common.CommandSender viewer) {
+        return getName();
     }
 }
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index 2c1b591a..2d1532f1 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -9,12 +9,12 @@ import java.util.UUID;
 
 import net.minecraft.server.*;
 
+import network.stratus.common.CommandSender;
 import org.bukkit.Chunk;
 import org.bukkit.EntityEffect;
 import org.bukkit.Location;
 import org.bukkit.Server;
 import org.bukkit.World;
-import org.bukkit.command.CommandSender;
 import org.bukkit.craftbukkit.CraftServer;
 import org.bukkit.craftbukkit.CraftWorld;
 import org.bukkit.event.entity.EntityDamageEvent;
@@ -645,6 +645,56 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         perm.setOp(value);
     }
 
+    @Override
+    public void setFakeName(CommandSender viewer, String name) {
+
+    }
+
+    @Override
+    public String getFakeName(CommandSender viewer) {
+        return getName();
+    }
+
+    @Override
+    public boolean hasFakeName(CommandSender viewer) {
+        return false;
+    }
+
+    @Override
+    public void clearFakeNames() {
+
+    }
+
+    @Override
+    public void setFakeDisplayName(CommandSender viewer, String name) {
+
+    }
+
+    @Override
+    public String getFakeDisplayName(CommandSender viewer) {
+        return getName();
+    }
+
+    @Override
+    public boolean hasFakeDisplayName(CommandSender viewer) {
+        return false;
+    }
+
+    @Override
+    public void clearFakeDisplayNames() {
+
+    }
+
+    @Override
+    public String getPlayerListName(CommandSender viewer) {
+        return getName();
+    }
+
+    @Override
+    public String getDisplayName(CommandSender viewer) {
+        return getName();
+    }
+
     // Spigot start
     private final Spigot spigot = new Spigot()
     {
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftMinecartCommand.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftMinecartCommand.java
index 1498bec6..db2f9ac4 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftMinecartCommand.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftMinecartCommand.java
@@ -56,11 +56,6 @@ public class CraftMinecartCommand extends CraftMinecart implements CommandMineca
         return ((EntityMinecartCommandBlock) getHandle()).getCommandBlock().getName();
     }
 
-    @Override
-    public String getName(org.bukkit.command.CommandSender viewer) {
-        return this.getName();
-    }
-
     @Override
     public boolean isOp() {
         return true;
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 442c8119..f9e98970 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -28,6 +28,7 @@ import net.md_5.bungee.api.chat.BaseComponent;
 import net.minecraft.server.*;
 import net.minecraft.server.PacketPlayOutTitle.EnumTitleAction;
 
+import network.stratus.common.CommandSender;
 import org.apache.commons.lang.Validate;
 import org.apache.commons.lang.NotImplementedException;
 import org.bukkit.*;
@@ -37,7 +38,6 @@ import org.bukkit.Statistic;
 import org.bukkit.Material;
 import org.bukkit.Statistic.Type;
 import org.bukkit.World;
-import org.bukkit.command.CommandSender;
 import org.bukkit.configuration.serialization.DelegateDeserialization;
 import org.bukkit.conversations.Conversation;
 import org.bukkit.conversations.ConversationAbandonedEvent;
@@ -92,7 +92,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
     private final Map<CommandSender, String> fakeNames = new WeakHashMap<CommandSender, String>();
     private final Map<CommandSender, String> fakeDisplayNames = new WeakHashMap<CommandSender, String>();
-    private final Map<CommandSender, Skin> fakeSkins = new WeakHashMap<CommandSender, Skin>();
+    private final Map<org.bukkit.command.CommandSender, Skin> fakeSkins = new WeakHashMap<>();
 
     public CraftPlayer(CraftServer server, EntityPlayer entity) {
         super(server, entity);
@@ -295,17 +295,17 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     }
 
     @Override
-    public boolean hasFakeSkin(CommandSender viewer) {
+    public boolean hasFakeSkin(org.bukkit.command.CommandSender viewer) {
         return viewer != null && this.fakeSkins.containsKey(viewer);
     }
 
     @Override
-    public Skin getFakeSkin(CommandSender viewer) {
+    public Skin getFakeSkin(org.bukkit.command.CommandSender viewer) {
         return viewer == null ? null : this.fakeSkins.get(viewer);
     }
 
     @Override
-    public void setFakeSkin(CommandSender viewer, Skin newSkin) {
+    public void setFakeSkin(org.bukkit.command.CommandSender viewer, Skin newSkin) {
         this.setFakeNameAndSkin(viewer, null, newSkin, false, true);
     }
 
@@ -315,7 +315,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     }
 
     @Override
-    public Skin getSkin(CommandSender viewer) {
+    public Skin getSkin(org.bukkit.command.CommandSender viewer) {
         if(this.hasFakeSkin(viewer)) {
             return this.getFakeSkin(viewer);
         } else {
@@ -385,15 +385,15 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
     @Override
     public void setFakeName(CommandSender viewer, String name) {
-        this.setFakeNameAndSkin(viewer, name, null, true, false);
+        this.setFakeNameAndSkin((org.bukkit.command.CommandSender) viewer, name, null, true, false);
     }
 
     @Override
-    public void setFakeNameAndSkin(CommandSender viewer, String name, Skin skin) {
+    public void setFakeNameAndSkin(org.bukkit.command.CommandSender viewer, String name, Skin skin) {
         this.setFakeNameAndSkin(viewer, name, skin, true, true);
     }
 
-    private void setFakeNameAndSkin(CommandSender viewer, String newName, Skin newSkin, boolean nameChanged, boolean skinChanged) {
+    private void setFakeNameAndSkin(org.bukkit.command.CommandSender viewer, String newName, Skin newSkin, boolean nameChanged, boolean skinChanged) {
         String oldName = this.fakeNames.get(viewer);
         nameChanged = nameChanged && !Objects.equal(oldName, newName);
         Skin oldSkin = this.fakeSkins.get(viewer);
-- 
2.20.1

